### PR TITLE
add complete callback form

### PR DESCRIPTION
**Pull request by rik86future**
- Adds the new Complete screen, this screen fetches the data from snapshot to populate form **notes**, it also sets the "**initial callback completed**" to Yes and sets the value for the "**follow up callback**" 

image

**Please note**: a callback required Date would be ideal, but we have no backing data to store it, and it was not part of this iteration anyway. But I strongly believe further improvement on the follow up callback is necessary because the Callback table will confuse the Agents, as it does not say when the follow up is needed for

### CODE REVIEW

File: controllers/help-requests/help_requests.controller.js [hunk 1]
```diff
@@ -260,32 +293,57 @@ module.exports = {
...
-                return next(error);
+                // prepare partial update
+                await HelpRequestsService.getHelpRequest(req.params.id)
+                  .then(originalRecord => {
+                      // keep the original record and update only case notes and callback
+                      let updateRequest = originalRecord
+                      updateRequest.NewCaseNote = query.NewCaseNote
+                      updateRequest.callback_required = query.callback_required
+                      updateRequest.initial_callback_completed = 'yes'
+                      handleUpdate(req, res, updateRequest, userName)
```
Comment (rik86future) [inline controllers/help-requests/help_requests.controller.js:340]: this is obviously inefficient, a new API endpoint to patch the data would be preferred, there is no API change as yet. I'll add a card to backlog

Comment (kjdchapman) [reply controllers/help-requests/help_requests.controller.js:343]: Good call on adding this to the backlog.

I think this is far below the things we flat can't do at the moment, though (like adding multiple case notes to a case).

**Review by kjdchapman — APPROVED** [review]
LGTM! :rocket: 

I am worried about how much change we're introducing to this app without automated testing. This might bite us soon.

File: controllers/help-requests/help_requests.controller.js (hunk 1 above)
Comment (rik86future) [reply controllers/help-requests/help_requests.controller.js:343]: I added that to the backlog too

**Pull request merged**